docs/develop/python/message-passing.mdx

```diff
@@ -431,113 +474,121 @@ class MyWorkflow:
 ```
 
-## Troubleshooting
-
-See [Exceptions in message handlers](/encyclopedia/workflow-message-passing#exceptions) for a non–Python-specific introduction.
-
-As a client, when you send a Signal, Update, or Query to a Workflow, the following two errors are always possible:
-
-1. **The client can't contact the server**
+See [Finishing handlers before the Workflow completes](/encyclopedia/workflow-message-passing#finishing-message-handlers) for more information.
 
-    You'll get an [`temporalio.service.RPCError`](https://python.temporal.io/temporalio.service.RPCError.html) on which the `status` attribute is [`RPCStatusCode`](https://python.temporal.io/temporalio.service.RPCStatusCode.html) `UNAVAILABLE` (after some retries; see the `retry_config` argument to [`Client.connect`](https://python.temporal.io/temporalio.client.Client.html#connect)).
+## Message handler troubleshooting {#message-handler-troubleshooting}
 
-1. **The workflow does not exist**
+When sending a Signal, Update, or Query to a Workflow, your Client might encounter the following errors:
 
-    You'll get an [`temporalio.service.RPCError`](https://python.temporal.io/temporalio.service.RPCError.html) on which the `status` attribute is [`RPCStatusCode`](https://python.temporal.io/temporalio.service.RPCStatusCode.html) `NOT_FOUND`.
+- **The client can't contact the server**:
+  You'll receive a [`temporalio.service.RPCError`](https://python.temporal.io/temporalio.service.RPCError.html) on which the `status` attribute is [`RPCStatusCode`](https://python.temporal.io/temporalio.service.RPCStatusCode.html) `UNAVAILABLE` (after some retries; see the `retry_config` argument to [`Client.connect`](https://python.temporal.io/temporalio.client.Client.html#connect)).
 
-### Problems when sending a Signal
+- **The workflow does not exist**:
+  You'll receive an [`temporalio.service.RPCError`](https://python.temporal.io/temporalio.service.RPCError.html) exception on which the `status` attribute is [`RPCStatusCode`](https://python.temporal.io/temporalio.service.RPCStatusCode.html) `NOT_FOUND`.
 
-For Signal, these `RPCError`s are the only types of exception that will result from the request.
+See [Exceptions in message handlers](/encyclopedia/workflow-message-passing#exceptions) for a non–Python-specific discussion of this topic.
 
-The situation is different for Query and Update since for these the client waits for a response from the Worker, and so the Client may get an exception indicating that something went wrong while the handler was being executed by the Workflow Worker.
+### Problems when sending a Signal {#signal-problems}
 
-### Problems when sending an Update
+When using Signal, the only exception that will result from your requests is `RPCError`.
 
-1. **There is no Workflow Worker polling the Task Queue**
+For Queries and Updates, the client waits for a response from the Worker.
+If an issue occurs during the handler Execution by the Worker, the client may receive an exception.
```

Comment:
This phrasing slightly loses the sense that the following errors are in addition to the `RPCError`s above that all handlers may experience in the initial (pre-worker) part of the request lifecycle.

Reply:
Working on this on-thread